### ISSUE TITLE
Adding support for modifying groups

### ISSUE THREAD

**Issue by samsworldofno** [jrallison/rollout_ui#4]
Hi,

I'm interested in adding support for modifying groups - particularly adding/removing users from groups, and maybe also being able to add/delete groups, although this is much less important. Have their been any discussions about how this functionality might be achieved already? I was thinking of adding a second page that would list groups and allow their modification - which would also mean adding some navigation to the layout.

Our specific use case is the Sinatra app, so I'd probably look at adding these features there in the first instance (as I can see the server's views are managed separately from the rails engine). Am I right in thinking test coverage for this area is low at the moment? If so, and I were to add some, is rspec + capybara the preferred method, or would cucumber be better?

Basically, any guidance on how this contribution would 'fit' would be happily received.

Cheers,

Sam

**Comment by jrallison** [comment]
Hey Sam,

Thanks for reaching out.

**Re: groups**
My inital take was groups are application specific and usually defined in ruby with access to the user model.  I didn't want to get into eval'ing ruby code in this gem.  However, if you view groups as just a collection of ids, then a simple version of that could be useful.  Having a separate "group management" page with simple navigation sounds like a solid approach.

If you feel comfortable fitting it into the current design, then go for it.  Otherwise, I can reach out to Holly, who did the original design, and see if she has any ideas.

**Re: testing**
My original goal was to deprecate the Sinatra app and just have the engine, but it seems that's less likely now as people are using RolloutUi without Rails and I'd like to keep supporting it.  Improvements I'd love to see are:
- better test coverage on the Sinatra app
- reduce the duplication of views/assets between the two if possible.

For new tests, rspec + capybara is the preferred method currently.

Looking forward to what you come up with!

Cheers,
John
